### PR TITLE
checking if url exist and implemented branches list selection

### PR DESCRIPTION
**Pull request by itsritiksingh**
1. it is now checking if repository url exits or not 
2. giving user list of branches to select rather than writing branches name

### CODE REVIEW

File: src/deploy.ts [hunk 1]
```diff
@@ -200,18 +200,29 @@ export const deployPackage = async (
...
-	const branch = await input('Enter branch name');
-	//TODO check if repo url is valid
+	const api = API(config.token as string, config.baseURL);
+
+	let branch;
 	const url = args['addrepo'];
 
-	info(`Deploying from ${url as string}...\n`);
+	try {
+		const { branches } = await api.branchList(url as string);
```
Comment (Creatoon) [inline src/deploy.ts:209]: I guess url will always be of string type as it is mentioned in the args type, so you don't need to pass ```url as string``` everywhere.

File: src/deploy.ts [hunk 2]
```diff
@@ -200,18 +200,29 @@ export const deployPackage = async (
...
 
-	const api = API(config.token as string, config.baseURL);
+		if (!branches.length) return error('Invalid Repository url');
```
Comment (Creatoon) [inline src/deploy.ts:211]:
```suggestion
		if (!branches.length) return error('Invalid Repository URL');
```

File: src/deploy.ts [hunk 3]
```diff
@@ -200,18 +200,29 @@ export const deployPackage = async (
...
+
+		branch = await listSelection(
+			branches,
+			'Select a container to get logs'
```
Comment (Creatoon) [inline src/deploy.ts:215]: Change the text, you are asking for logs selection but you want the user to select a specific branch

File: src/deploy.ts (hunk 1 above)
Comment (itsritiksingh) [reply src/deploy.ts:209]: No type of url was <string | undefined> which was linter was not allowing to be a argument which should be a string type only

Comment (Creatoon) [reply src/deploy.ts:209]: So wrap the whole things like this ```if(typeof url  !== 'undefined')```
Do this only you think writing this way can reduce ```as``` from most of the places as code will look cleaner then.

Comment (itsritiksingh) [reply src/deploy.ts:209]: no did not solve this problem
i have tried

**Comment by itsritiksingh** [comment]
i'm deleting this pr since it has many faults

**Pull request closed without merging**